File: Semester 3/OOP/Week10/Q3.java
class Node<T>
{
	T data;
	Node<T> next;
	Node(T d)
	{
		data=d;
		next=null;
	}
}
class List<T>
{
	Node<T> head;
	List()
	{
		head=null;
	}
	void insertFront(T x)
	{
		Node<T> temp = new Node<T>(x);
		if (head==null) head=temp;
		else 
		{
			temp.next=head;
			head=temp;
		}
	}
	void deleteFront()
	{	
		if (head==null) System.out.println("Cannot Delete!");
		else 
		{
			T value = head.data;
			head = head.next;
			System.out.println("\nDeleted:  " + value);			
		}
	}
	void display()
	{	
		if (head==null) System.out.println("Empty");
		for (Node<T> t = head; t!=null; t=t.next)
		{
			System.out.print(t.data+ " ");
		}
	}
}
class Q3
{	
	public static void main(String args[])
	{
		List<Integer> intList = new List<>();
		List<Double> doubList = new List<>();
		// Integer x = 10; Integer y = 20;
		// Double z = 10.5;
		intList.insertFront(10); intList.insertFront(20);
		System.out.print("Integer List: "); intList.display();
		intList.deleteFront();
		doubList.insertFront(10.5);
		System.out.print("Double List: "); doubList.display();	
		doubList.deleteFront();
		doubList.deleteFront();
	}

}

File: Semester 3/DSA/Week09/q3.c
#include<stdio.h>
#include<stdlib.h>
typedef struct node *Node;
typedef struct node{
    char data;
    Node next;
}LL;
Node getNode(){
    Node temp = (Node)malloc(sizeof(LL));
    return temp;
}
void InsertRear(Node* List,int ele){
    Node head = *List;
    Node rear;
    Node temp = getNode();
    temp->data = ele;
    temp->next = head;
    if(*List == NULL){
        *List = temp; head = *List;
        (*List)->next = head; return;
    } rear = head->next;
    while(rear->next != head)rear = rear->next;
    rear->next = temp;
}
void display(Node List){
    Node head = List;
    printf("%c ",head->data);
    Node temp = head->next;
    while(temp != head){
        printf("%c ",temp->data); temp = temp->next;
}}
void sort(Node* List){
    Node head = *List, current = *List, next = NULL;
    char x;
    while(current->next != head){
        next = current->next;
        while(next != head){
            if(current->data > next->data){
                x = current->data; current->data = next->data;
                next->data = x;
            } next = next->next;
        } current = current->next;
}}
void merge(Node* List1,Node List2){
    Node First = *List1, Second = List2, rear = First->next, temp;
    while(rear->next != First)rear = rear->next;
    temp = getNode();
    temp->data = List2->data; temp->next = First;
    rear->next = temp; rear = rear->next;
    List2 = List2->next;
    while(List2 != Second){
        temp = getNode();
        temp->data = List2->data; temp->next = First;
        rear->next = temp; rear = rear->next;
        List2 = List2->next;
}}
void main(){
    Node List1 = NULL, List2 = NULL;
    InsertRear(&List1,'d'); InsertRear(&List1,'e'); InsertRear(&List1,'f');
    display(List1); printf("\n");
    InsertRear(&List2,'a'); InsertRear(&List2,'b'); InsertRear(&List2,'c');
    display(List2); printf("\n");
    merge(&List1,List2); sort(&List1); display(List1);
}